### DISCUSSION TITLE
Payment at the order page in the customers order history

### DISCUSSION THREAD

**Discussion by karopka** [PrestaShop/PrestaShop#40190] [Ideas]
It may happen that the buyer wants to choose another payment method (or the seller wants to offer another payment method) after placing the order. For example, a cash payment was chosen, and then the buyer decided to pay by card on the site. But the order has already been submited.

It would be great to add this feature on the Order History page.

As I see it: the seller changes the status of the order to a previously defined one and the 'Pay' button appears in the order page.

**Comment by khouloudbelguith** [comment]
HI @karopka,

Thanks for your report.
In our [Marketplace](https://addons.prestashop.com/en) you can find a lot of modules to edit payments methods.
For example, you can find this [module](https://addons.prestashop.com/en/order-management/18850-order-edit-change-and-modify-existing-order.html).
I close the issue, feel free to open a new one when needed.

**Comment by karopka** [comment]
Thats modules do not offer payment mechanism for a Customer. You just can change payment method at admin area but nothing happens at frontend. If I switch to online pay by credit card there is no button appears at Customers account to do a payment. 

Or I missed something.

**Comment by khouloudbelguith** [comment]
@karopka, thanks for this clarifications.
@marionf, @colinegin, can we add this feature to the core PrestaShop, on the front end?
In the order history details, can we add an options to edit the payment method after a complete order?
Thanks!

**Comment by karopka** [comment]
Not exactly this feature. I have asked about ability to invoke payment process not only at cart on checkout but in the order history if administrator or manager switch order status for special status for this after conversation with Customer.

The example is Customer wish to change payment method after order submiting.

**Comment by colinegin** [comment]
Hello @karopka, 

Thanks for submitting your idea ! It seems complicated to me as it needs to check the payment methods available on the website and define which ones can be switched with other methods. What do you think ?

Anyway, we are always glad to get new ideas, so I've just posted your feature idea on http://feedback.prestashop.com/forums/387864-prestashop-1-7-x, to help us prioritize next developments ! Do not hesitate to add comments, as soon as it will be displayed.

Have a nice day,

Coline

**Comment by hibatallahAouadni** [comment]
Ping @marionf I can see that the issue is approuved by the PM but without status :confused: so, what's the status of this issue? is it **`To Do`** or **`TBS`**?
